tbx/project_styleguide/templates/patterns/molecules/streamfield/blocks/markdown_block.html

```diff
@@ -1,5 +1,8 @@
-{% load wagtailmarkdown %}
+{% load wagtailmarkdown static %}
+
 <div class="grid__markdown markdown-block">
+    <link rel="stylesheet" type="text/css" href="{% static 'css/codehilite.css' %}">
```

Comment:
q: if we include a bunch of markdown or gists on a page, then we'd have multiple style declarations. Are browsers smart enough not to download them again and again?

Reply:
I assume they'd load it once and then any subsequent requests would be taken from cache. But i'm wondering if the browser would still have to parse the files regardless of that so ideally we'd only add one script tag in. 
Is there a better way to handle this on the BE?

Reply:
in the BE, you'd loop from all the body blocks and check if whether the given blocks are there, and add some vars to the context then conditionally load the stylesheets once. Will share a snippet or diff shortly

Reply:
Could we add a custom filter:
```
@register.filter(name="has_raw_html_block")
def has_raw_html_block(value):
    if isinstance(value, StreamValue):
        return any(type(item.block) is blocks.RawHTMLBlock for item in value)
    return False
```

and then in the template

```
{% block extra_css %}
    {{ block.super }}
    {% if page.body|has_raw_html_block %}
        <link rel="stylesheet" type="text/css" href="{% static 'css/gist.css' %}">
    {% endif %}
{% endblock %}
```
?

Reply:
https://github.com/torchbox/torchbox.com/pull/295/commits/7386e6ac3b80583e2a4eee28f830c92e154ef858 👀

Reply:
That will work. with the raw html block you want to check that "https://gist.github.com" is in the value too, so it only loads for raw html blocks with gists

```python
@register.filter(name="has_raw_html_block")
def has_raw_html_block(value):
    if not isinstance(value, StreamValue):
        return False
    
    for block in value._raw_data:
        if block["type"] == "raw_html" and "https://gist.github.com" in block["value"]:
            return True
    return False
```

The same approach would apply for the markdown block.